packages-exp/app-check-compat/src/index.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import firebase, {
+  _FirebaseNamespace,
+  FirebaseApp
+} from '@firebase/app-compat';
+import { name, version } from '../package.json';
+import {
+  Component,
+  ComponentContainer,
+  ComponentType,
+  InstanceFactory
+} from '@firebase/component';
+import { AppCheckService } from './service';
+import { FirebaseAppCheck } from '../../../packages/app-check-types';
+
+declare module '@firebase/component' {
+  interface NameServiceMapping {
+    'app-check-compat': AppCheckService;
+  }
+}
+
+const factory: InstanceFactory<'app-check-compat'> = (
+  container: ComponentContainer
+) => {
+  // Dependencies
+  const app = container.getProvider('app-compat').getImmediate();
+  const AppCheckServiceExp = container
+    .getProvider('app-check-exp')
+    .getImmediate();
+
+  return new AppCheckService(app as FirebaseApp, AppCheckServiceExp);
+};
+
+export function registerAppCheck(): void {
+  (firebase as _FirebaseNamespace).INTERNAL.registerComponent(
+    new Component(
+      'app-check-compat',
```

Comment:
```suggestion
      'appCheck-compat',
```

Need to name it this way so that people can call `firebase.appCheck()`

Reply:
Fixed.